### PR TITLE
Fetch mobile

### PR DESCRIPTION
**Pull request by arth-shukla**
---

# ms2 mobile base integration + other fixes

---

- fetch mobile base
- fixed to grasping code
- utils, controllers, sim outputs, warnings, etc
- joint base vel controller fixes are cpu-sim specific and need to be updated with gpu sim support in the future

---
---
---

# ms3 commit text

---

**DRAFT** awaiting testing w/ sapien version from main branch

below are details on the testing done on the fetch robot to make sure it still works properly after the hacky mobile joints are added

---

# stationary base tests

in the charts below, darker colors are stationary torso and lighter colors are mobile torso

## baseline

this is the same as the previous pr (no extra joints, no collision group changes)
baseline

## joints added as `type="fixed"`, ground collision group 3rd item set to `1 << 37`

one seed for mobile torso fails, the rest succeed
fixed-joint_floor-collision

## fixed joints with 0 mass/inertia, set third ground collision group

all seeds work
no-mass-inertia_fixed-joint_floor-collision

---

# mobile base tests

these all have floor collisions off with the extra prismatic/continuous joints added. "stationary" means the action is masked to remove control over that part of the robot

red: stationary torso, stationary base
blue: mobile torso, stationary base
green: mobile torso, mobile base

mobile-base-eval

the additional mobility results in some seeds seeds with more 'failure'/slow convergence

## stationary torso, stationary base

|  **Success** | **Failure**  |
|---|---|
| https://github.com/haosulab/ManiSkill3/assets/44527268/f2fc142f-e926-42bf-95f2-a5dd09ebe221 | https://github.com/haosulab/ManiSkill3/assets/44527268/e5629f3e-ce7e-4618-947a-794714f8e49a |

same as before, fails when controller reaches singularity

## mobile torso, stationary base

|  **Success** | **Failure**  |
|---|---|
| https://github.com/haosulab/ManiSkill3/assets/44527268/b3006022-ca97-4341-b91c-90f017f932f4 | https://github.com/haosulab/ManiSkill3/assets/44527268/80903507-b17b-44ae-9989-f6eeb510a62f |

success: same as before

failure: 1 seed which does not reach success, seemingly uses too-broad torso movements

## mobile torso, mobile base

|  **Success** | **Failure**  |
|---|---|
| https://github.com/haosulab/ManiSkill3/assets/44527268/eabadb6a-6bd9-452a-9e36-ec42ff94e25a | https://github.com/haosulab/ManiSkill3/assets/44527268/445dda9c-e3b3-4a4c-9775-fa5066da56fd |

success: note that, despite using end-effector delta **pos**, the robot uses its mobile base to achieve a different **pose** than normal and reach the goal

failure: 2 seeds with slower convergence, seemingly because they try to move the block by leveraging arm against table. one way to fix is to add collision penalty to reward

# next exps

1. collision penalty to avoid leveraging arm against table
2. test mobile manip spawning on different parts of table
3. overfit to hssd object, then run clustering to finetune/train new policies for each object

### CODE REVIEW

File: mani_skill2/agents/controllers/pd_base_vel.py [hunk 1]
```diff
@@ -18,14 +19,21 @@ def _initialize_action_space(self):
     def set_action(self, action: np.ndarray):
         action = self._preprocess_action(action)
 
+        # TODO (arth): add support for batched qpos and gpu sim
```
Comment (StoneT2000) [inline mani_skill2/agents/controllers/pd_base_vel.py:22]: good that you added the todos, otherwise we will all forget this XD

**Review by StoneT2000 — APPROVED** [review]
